src/cmd/pack.ts

```diff
@@ -16,26 +16,38 @@ import { Block } from "@ipld/unixfs";
 import { CAREncoderStream } from "ipfs-car";
 import { CID } from "multiformats/cid";
 
+import { mergeDeep } from "../contents.js";
 import { iterateFilesFromPathsWithSize } from "../files.js";
-import type { SuperManifest, UserMetadata } from "../manifest.d.js";
+import type {
+  SubManifest,
+  SuperManifest,
+  SuperManifestContentEntry,
+  UserMetadata,
+} from "../manifest.d.js";
+import parseBytes from "../parseBytes.js";
 import { createDirectoryEncoderStream } from "../unixfs.js";
 
 // Root CID written in CAR file header before it is updated with the real root CID.
 const placeholderCID = CID.parse(
   "bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi"
 );
 
+const currentSpecURL =
```

Comment:
Maybe gilding the lily a bit but given that the version needs to be known in a handful of places wouldn't it be better to just define the version and then have a generator for all the strings that come out of it?

Reply:
Yes, I plan to extract all the manifest stuff out into a class.  We currently waste memory by keeping copies of the sub manifests around after they are no longer needed.  I plan to improve that situation in the class.  Would it be acceptable to do that in a separate PR please?

Reply:
Sure